packages/react-router/lib/server-runtime/server.ts

```diff
@@ -230,6 +230,10 @@ export const createRequestHandler: CreateRequestHandlerFunction = (
       );
     }
 
+    if (!response) {
+      return new Response("Unknown Server Error", { status: 500 });
```

Comment:
We can probably run this through the same utility we use elsewhere

```suggestion
      let error = new Error('Unhandled request')
      return returnLastResortErrorResponse(error, serverMode);
```